aws-serverless-java-container-jersey/src/main/java/com/amazonaws/serverless/proxy/jersey/JerseyServletResponseWriter.java

```diff
@@ -122,14 +122,8 @@ public void commit() {
 
 
     public void failure(Throwable throwable) {
-        try {
-            log.error("failure", throwable);
-            jerseyLatch.countDown();
-            servletResponse.flushBuffer();
-        } catch (IOException e) {
-            log.error("Could not fail response", e);
-            throw new InternalServerErrorException(e);
-        }
+        log.error("failure", throwable);
+        throw new InternalServerErrorException("Jersey failed to process request", throwable);
```

Comment:
Wondering what will happen to the latch in this case. The behavior seems correct as far as the outcome (what we return to the client). I'll add a TODO list item to verify that we don't leak anything in this case. I'll take care of it before the next release

Reply:
Yeah, I tested that `latch.await()` is not called during unit tests but this might not be the case every time. Looking closer at the code it could cause a leak.  I took a look at Jersey's [`ApplicationHandler.java`](https://github.com/jersey/jersey/blob/master/core-server/src/main/java/org/glassfish/jersey/server/ApplicationHandler.java#L672-L676) they use a future and call `.completeExceptionally()`. Maybe the code should be refactored to use futures instead of a latch. Also I wonder how framework failures are handled by the other consumers of `LambdaContainerHandler.java`.

Reply:
I've created issue #276 to track this for the 2.0 release of the framework.